gammapy/astro/darkmatter/profiles.py

```diff
@@ -34,19 +34,22 @@ def scale_to_local_density(self):
         scale = (self.LOCAL_DENSITY / self(self.DISTANCE_GC)).to_value("")
         self.parameters["rho_s"].value *= scale
 
-    def _eval_squared(self, radius, separation):
-        """Squared density at given radius together with the substitution part"""
+    def _eval_substitution(self, radius, separation, squared):
+        """Density at given radius together with the substitution part"""
+        exponent = 2 if squared else 1
         return (
-            self(radius) ** 2
+            self(radius) ** exponent
             * radius
             / np.sqrt(radius**2 - (self.DISTANCE_GC * np.sin(separation)) ** 2)
         )
 
-    def integral(self, rmin, rmax, separation, ndecade):
-        r"""Integrate squared dark matter profile numerically.
+    def integral(self, rmin, rmax, separation, ndecade, squared):
+        r"""Integrate dark matter profile numerically.
```

Comment:
The same here for the docstring... Generally speaking, improving the API documentation can help a lot the Gammapy non-experts

Reply:
This already done a few lines down